File: backend/app/schemas/guide.py
from pydantic import BaseModel


class ObstacleRequest(BaseModel):
    frame: str  # base64 JPEG


class ObstacleResponse(BaseModel):
    warning: str
    severity: str  # "clear", "caution", "danger"
    audio_url: str
    provider: str = ""
    latency_ms: int = 0

File: mobile/src/types/guide.ts
export interface ObstacleWarning {
  warning: string;
  severity: "clear" | "caution" | "danger";
  timestamp: string;
}

// Guide WebSocket messages
export interface WSLocationUpdate {
  type: "location_update";
  lat: number;
  lng: number;
}

export interface WSNextStep {
  type: "next_step";
  step_index: number;
  instruction: string;
  audio_url: string;
}

export interface WSOffRoute {
  type: "off_route";
  message: string;
}

export interface WSArrived {
  type: "arrived";
  message: string;
}

export type WSGuideServerMessage = WSNextStep | WSOffRoute | WSArrived;
